tests/ui/function-stubbing-error/unsupported_resolutions.expected

```diff
@@ -0,0 +1,8 @@
+error: failed to resolve `<[char ; 10]>::foo`: Kani currently cannot resolve path including primitive types
```

Comment:
Somewhat random place for comment: do I understand correctly that no test is being added that demonstrates that the feature works? As far as I can tell, we only have tests demonstrating that name resolution fails?!

Reply:
That's exactly what this PR is doing. This is not yet adding a new feature or a new fix, hence only the failure case. This PR is the first step to implement the feature. In this PR, we are now able to correctly parse the user input for fully qualified path and primitive types, and distinguish between them. We still need to create a follow up PR to handle each case identified here.

Reply:
The PR name says "Extend name resolution to support qualified paths," so I was assuming that we can now successfully resolve some (qualified) paths that we weren't able to resolve before?

Reply:
Yes. We can resolve the names now. But we cannot use the resolution for stubbing or contracts yet.

Reply:
I edited the name of this PR so it is explicit that this is a partial fix. Sorry for the confusion.